src/mesh/voxel_cloud.rs

```diff
@@ -404,17 +388,13 @@ impl ScalarField {
                 .coords;
 
                 target_scalar_field.block_start += voxel_final_translation_vector;
-
-                // FIXME: @Optimization In some cases it might be not easy to
-                // determine the minimal needed bounding box due to the
-                // differences in the voxel sizes. Leaving the scalar field
-                // bigger should be also considered.
+                // FIXME: Consider not shrinking to fit because in most cases it
+                // won't be much larger than it has to be.
                 target_scalar_field.shrink_to_fit(volume_value_range);
 
                 return Some(target_scalar_field);
```

Comment:
I think sometimes the target scalar field may be larger than necessary, so `shrink_to_fit` may be appropriate. On the other hand, it may be even slower to shrink to fit and it may be smarter to keep the scalar field unnecessarily big. That applies to all functions shrinking the result scalar field. Maybe the shrinking could only happen in the operations and also only when there is too much obsolete space.

Reply:
I mean the shrink could be called explicitly rather than inside functions, that may or may not result in an unnecessarily large scalar field. Because maybe the caller wants to get the entire scalar field rather than its trimmed version.

Reply:
> Maybe the shrinking could only happen in the operations and also only when there is too much obsolete space.

Good idea!

Reply:
https://trello.com/c/2XSIdfuH/343-only-shrink-voxel-clouds-scalar-fields-to-fit-explicitly-in-operations